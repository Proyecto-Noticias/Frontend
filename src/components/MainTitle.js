import React, { useRef, useEffect } from "react";
import { TweenMax, Power3 } from "gsap";
import { Link } from "react-router-dom";
import InstallPWA from "../components/installPWA";

const MainTitle = () => {
  let titleWithAnimation = useRef(null);
  useEffect(() => {
    TweenMax.to(titleWithAnimation, 1, {
      opacity: 1,
      y: -20,
      ease: Power3.easeOut,
    });
  }, []);

  return (
    <>
      <div
        ref={(el) => {
          titleWithAnimation = el;
        }}
        className="main__container--login">
        <h1 className="main__container--title">
          News with automated web extraction
        </h1>
        <p className="main__container--title-description">
          We use Artificial Intelligence to classify the news according to its
          positivism or negativity
        </p>
        <Link aria-label="Learn more" to="/blog">
          <button tabIndex="-1" className="main__container--btn" type="button">
            Learn more
          </button>
        </Link>
        <InstallPWA />
      </div>
    </>
  );
};

export default MainTitle;
