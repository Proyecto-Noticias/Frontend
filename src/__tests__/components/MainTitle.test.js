import React from 'react';
import { shallow } from "enzyme";
import MainTitle from '../../components/MainTitle';


describe('Test of <MainTitle />', () => {
  
  
  const wrapper = shallow( <MainTitle /> )  
  
  test('should show the component correctly', () => {   
    expect ( wrapper ).toMatchSnapshot()
  })

  test('should contain className main__container--login', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')

    expect( className.includes('main__container--login')).toBe( true )
  })

  test('should contain className main__container--title', () => {
    const h1 = wrapper.find('h1')
    const className = h1.prop('className')

    expect( className.includes('main__container--title')).toBe( true )
  })

  test('should contain className main__container--title-description', () => {
    const p = wrapper.find('p')
    const className = p.prop('className')

    expect( className.includes('main__container--title-description')).toBe( true )
  })
  
})