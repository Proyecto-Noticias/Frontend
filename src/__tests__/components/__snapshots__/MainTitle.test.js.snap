// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <MainTitle /> should show the component correctly 1`] = `
<Fragment>
  <div
    className="main__container-login"
  >
    <h1
      className="main__title"
    >
      News with automated web extraction
    </h1>
    <p
      className="main__title-description"
    >
      We use Artificial Intelligence to classify the news according to its positivism or negativity
    </p>
    <Link
      to="/blog"
    >
      <button
        aria-label="Learn more"
        className="main__btn"
        type="button"
      >
        Learn more
      </button>
    </Link>
  </div>
</Fragment>
`;
